packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -198,113 +199,59 @@ def _get_field(mcs, cls: Type, field_name: str) -> Any:
+    id_: str = field(default_factory=lambda: uuid.uuid4().hex, init=False)
 
     @property
-    def sender(self) -> str:
-        """Get the sender."""
-        return self._sender
+    def data(self) -> Dict[str, Any]:
+        """Data"""
+        excluded = ["sender", "round_count", "id_", "transaction_type"]
+        return {k: v for k, v in asdict(self).items() if k not in excluded}
 
+    # TODO: refactor - these methods are not strictly needed
     @property
-    def round_count(self) -> int:
-        """Get the round count."""
-        return self._round_count
-
-    @round_count.setter
-    def round_count(self, round_count: int) -> None:
-        """Set the round count."""
-        self._round_count = round_count
-
-    def encode(self) -> bytes:
-        """Encode the payload."""
-        return DictProtobufStructSerializer.encode(self.json)
-
-    @classmethod
-    def decode(cls, obj: bytes) -> "BaseTxPayload":
-        """Decode the payload."""
-        return cls.from_json(DictProtobufStructSerializer.decode(obj))
+    def json(self) -> Dict[str, Any]:
+        """Json"""
+        data = asdict(self)
+        data["transaction_type"] = str(data["transaction_type"])
+        return data
 
     @classmethod
     def from_json(cls, obj: Dict) -> "BaseTxPayload":
         """Decode the payload."""
         data = copy(obj)
         transaction_type = str(data.pop("transaction_type"))
+        round_count, id_ = data.pop("round_count"), data.pop("id_")
         payload_cls = _MetaPayload.transaction_type_to_payload_cls[transaction_type]
-        return payload_cls(**data)
-
-    @property
-    def json(self) -> Dict:
-        """Get the JSON representation of the payload."""
-        return dict(
-            transaction_type=str(self.transaction_type),
-            id_=self.id_,
-            sender=self.sender,
-            round_count=self.round_count,
-            **self.data,
-        )
-
-    @property
-    def data(self) -> Dict:
-        """
-        Get the dictionary data.
-
-        The returned dictionary is required to be used
-        as keyword constructor initializer, i.e. these two
-        should have the same effect:
-
-            sender = "..."
-            some_kwargs = {...}
-            p1 = SomePayloadClass(sender, **some_kwargs)
-            p2 = SomePayloadClass(sender, **p1.data)
-
-        :return: a dictionary which contains the payload data
-        """
-        return {}
+        payload = payload_cls(**data)  # type: ignore
+        object.__setattr__(payload, "round_count", round_count)
+        object.__setattr__(payload, "id_", id_)
+        return payload
 
     def with_new_id(self) -> "BaseTxPayload":
         """Create a new payload with the same content but new id."""
-        return type(self)(
-            self.sender, id_=uuid.uuid4().hex, round_count=self.round_count, **self.data
-        )
+        new = type(self)(sender=self.sender, **self.data)  # type: ignore
```

Comment:
```suggestion
        new = type(self)(sender=self.sender, **self.data)  # type: ignore
```
is missing the `round_count` and `id_`!

Reply:
- `id_`
https://github.com/valory-xyz/open-autonomy/blob/7a18eb4479a26891fe3e807abebea4fd4d5f9e32/packages/valory/skills/abstract_round_abci/base.py#L206
- `round_count`
https://github.com/valory-xyz/open-autonomy/blob/7a18eb4479a26891fe3e807abebea4fd4d5f9e32/packages/valory/skills/abstract_round_abci/base.py#L234-L238